File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/DisplayScoreboardPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public final record DisplayScoreboardPacket(byte position, String scoreName) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeByte(position);
        writeString(os, scoreName);
    }

    @Override
    public byte getId() {
        return 0x3D;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/DisplayScoreboardPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class DisplayScoreboardPacket(val position: Byte, val scoreName: String) : Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeByte(position.toInt())
        os.writeString(scoreName)
    }

    override val id: Byte
        get() = 0x3D
}
